shared/chat/inbox/container/remote.js

```diff
@@ -13,10 +14,45 @@ const getMetaMap = (state: Container.TypedState) => [state.chat2.metaMap, state]
 const createShallowEqualSelector = Container.createSelectorCreator(Container.defaultMemoize, shallowEqual)
 
 // Get conversations
-const getMetas = Container.createSelector([getMetaMap], ([metaMap, state]) => [
-  metaMap.filter((meta, id) => Constants.isValidConversationIDKey(id)),
-  state,
-])
+const getMetas = Container.createSelector([getMetaMap], ([metaMap, state]) => {
+  const _username = state.config.username
+  /*
+    sort priority is determined by the following criteria:
+    - adhoc (DM's) chats come first
+    - mentioned channels
+    - recent activity in channel
+  */
+
+  const prioritizedMetas = metaMap.sort((a, b) => {
+    if (a.teamType === 'adhoc' && b.teamType === 'adhoc') {
+      // chats first
+      if (
+        Constants.getHasUnread(state, b.conversationIDKey) ||
+        Constants.getHasUnread(state, a.conversationIDKey)
+      ) {
+        // sort unread message by time
+        return b.timestamp - a.timestamp
+      }
+    } else if (a.teamType === 'big' || b.teamType === 'big') {
+      // channels
+      // check if user has been mentioned in the message
+      if (a.snippet.indexOf(`@${_username}`) === 0 || b.snippet.indexOf(`@${_username}`) === 0) {
+        return b.timestamp - a.timestamp
+      } else if (
+        (a.snippet.toLowerCase().indexOf('you') === 0 &&
```

Comment:
Hmm I'd guess there would be a better way to check if the latest message is sent by `you`. Same for the @-mentions above, by looking at stuff from the store.

Reply:
Let me look through the store and see how it's looking